*** include/constraint_solver/constraint_solveri.h.orig	2017-02-09 13:36:10.799263006 +0100
--- include/constraint_solver/constraint_solveri.h	2017-02-09 13:42:27.329956476 +0100
***************
*** 63,68 ****
--- 63,69 ----
  #include "base/integral_types.h"
  #include "base/logging.h"
  #include "base/sysinfo.h"
+ #include "base/casts.h"
  #include "base/timer.h"
  #include "base/join.h"
  #include "base/sparse_hash.h"
***************
*** 252,260 ****
  
  inline uint64 Hash1(void* const ptr) {
  #if defined(ARCH_K8) || defined(__powerpc64__) || defined(__aarch64__)
!   return Hash1(reinterpret_cast<uint64>(ptr));
  #else
!   return Hash1(reinterpret_cast<uint32>(ptr));
  #endif
  }
  
--- 253,261 ----
  
  inline uint64 Hash1(void* const ptr) {
  #if defined(ARCH_K8) || defined(__powerpc64__) || defined(__aarch64__)
!   return Hash1(bit_cast<uint64_t>(ptr));
  #else
!   return Hash1(bit_cast<uint32_t>(ptr));
  #endif
  }
  
